content/en/blog/_posts/2025-07-28-kubernetes-v1.34-sneak-peek.md

```diff
@@ -0,0 +1,97 @@
+This effort is complemented by a parallel enhancement, [KEP-647](https://kep.k8s.io/647), which brings the same tracing capabilities to the Kubernetes API server. Together, these enhancements provide a more unified, end-to-end view of events, simplifying the process of pinpointing latency and errors from the control plane down to the node. These features have matured through the official Kubernetes release process. [KEP-2831](https://kep.k8s.io/2831) was introduced as an alpha feature in v1.25, while [KEP-647](https://kep.k8s.io/647) debuted as alpha in v1.22. Both enhancements were promoted to beta together in the v1.27 release. Looking forward, Kubelet Tracing ([KEP-2831](https://kep.k8s.io/2831)) is now targeting graduation to a stable in the upcoming v1.34 release.
+
+### `PreferSameZone` and `PreferSameNode` traffic distribution for Services
+
+The `spec.trafficDistribution` field within a Kubernetes [Service](/docs/concepts/services-networking/service/) allows users to express preferences for how traffic should be routed to Service endpoints.  
+[KEP-3015](https://kep.k8s.io/3015) deprecates `PreferClose` and introduces two additional values: `PreferSameZone` and `PreferSameNode`. `PreferSameZone` is equivalent to the current `PreferClose`. `PreferSameNode` prioritizes sending traffic to endpoints on the same node as the client.  
+This feature was introduced in v1.33 behind the `PreferSameTrafficDistribution` feature gate. It is targeting graduation to beta in v1.34 with its feature gate enabled by default.
+
+### Support for KYAML: a Kubernetes dialect of YAML
+
+KYAML is a safer and less ambiguous YAML subset designed specifically for Kubernetes. You can use KYAML for writing manifests and Helm charts, and request KYAML output from `kubectl` (as in `kubectl get -o kyaml ...`).
+
+KYAML addresses specific challenges with both YAML and JSON. YAML's significant whitespace requires careful attention to indentation and nesting, while its optional string-quoting can lead to unexpected type coercion (for e.g.: ["The Norway Bug"](https://hitchdev.com/strictyaml/why/implicit-typing-removed/)). Meanwhile, JSON lacks comment support and has strict requirements for trailing commas and quoted keys.  
+
+[KEP-5295](https://kep.k8s.io/5295) introduces KYAML, which addresses the problems by:
```

Comment:
fyi @lmktfy, to address https://github.com/kubernetes/website/pull/51651#discussion_r2226153074
```suggestion
[KEP-5295](https://kep.k8s.io/5295) introduces KYAML, which tries to address the most significant problems by:
```